src/H5Dmpio.c

```diff
@@ -5601,256 +5350,128 @@ H5D__mpio_get_chunk_insert_info_types(MPI_Datatype *contig_type, hbool_t *contig
-#if H5_CHECK_MPI_VERSION(3, 0)
-                        if (MPI_SUCCESS != (mpi_code = MPI_Get_address(chunk_list[i].buf, &base_buf)))
-                            HMPI_GOTO_ERROR(FAIL, "MPI_Get_address failed", mpi_code)
-#else
-                        base_buf = (MPI_Aint)chunk_list[i].buf;
-#endif
-
-                        base_offset = chunk_list[i].chunk_current.offset;
-                    }
-                }
-
-                /* Set convenience pointer for current chunk block */
-                chunk_block =
-                    (op_type == H5D_IO_OP_READ) ? &chunk_list[i].chunk_current : &chunk_list[i].chunk_new;
-
-                /*
-                 * Set the current chunk entry's offset in the file, relative to
-                 * the first chunk entry
-                 */
-                assert(H5_addr_defined(chunk_block->offset));
-                file_offset_array[chunk_count] = (MPI_Aint)(chunk_block->offset - base_offset);
-
-                /*
-                 * Ensure the chunk list is sorted in ascending ordering of
-                 * offset in the file
-                 */
-                if (chunk_count)
-                    assert(file_offset_array[chunk_count] > file_offset_array[chunk_count - 1]);
-
-                /* Set the current chunk entry's size for the I/O operation */
-                H5_CHECK_OVERFLOW(chunk_block->length, hsize_t, int);
-                length_array[chunk_count] = (int)chunk_block->length;
-
-                /*
-                 * Set the displacement of the chunk entry's chunk data buffer,
-                 * relative to the first entry's data buffer
-                 */
-#if H5_CHECK_MPI_VERSION(3, 1)
-                if (MPI_SUCCESS != (mpi_code = MPI_Get_address(chunk_list[i].buf, &chunk_buf)))
-                    HMPI_GOTO_ERROR(FAIL, "MPI_Get_address failed", mpi_code)
-
-                io_buf_array[chunk_count] = MPI_Aint_diff(chunk_buf, base_buf);
-#else
-                chunk_buf                 = (MPI_Aint)chunk_list[i].buf;
-                io_buf_array[chunk_count] = chunk_buf - base_buf;
-#endif
-
-                /*
-                 * Set last valid index in case only a single chunk will
-                 * be involved in the I/O operation
-                 */
-                last_valid_idx = i;
-
-                chunk_count++;
-            } /* end for */
+                            "couldn't allocate space for I/O buffers vector")
         }
 
         /*
-         * Create derived datatypes for the chunk list if this
-         * rank has any chunks to work on
+         * Since all I/O will be raw data, we can save on memory a bit by
+         * making use of H5FD_MEM_NOLIST to signal that all the memory types
+         * are the same across the I/O vectors
          */
-        if (chunk_count > 0) {
-            if (chunk_count == 1) {
-                int chunk_len;
+        if (NULL == (io_types = H5MM_malloc(2 * sizeof(*io_types))))
```

Comment:
Why allocate this dynamically?   An array of size 2 is no problem on the stack.

Reply:
Just a leftover from dynamically allocating all the arrays to be `iovec_count` in size and then later remembering I don't need to for the `io_types` array. Will update.